Adds user_permissions property and be_inherited matcher to the file resource for Windows. (#5775)
chef/inspec: docs-chef-io/content/inspec/resources/file.md

    its('product_version') { should eq '2.3.4' }

### user_permissions

> The `user_permissions` property returns the hash containing the list of users or groups and their file permissions on Windows. for e.g. `{ "NT AUTHORITY\\SYSTEM" => "FullControl", "NT AUTHORITY\\Authenticated Users" => "ReadAndExecute", "BUILTIN\\Administrators" => "FullControl" }`

Reviewer: ```suggestion
The `user_permissions` property returns a hash containing a list of users or groups and their file permissions on Windows. For example:
```

      it { should_not be_more_permissive_than('0644') }
      it { should be_more_permissive_than('0000') }
    end

> ### `be_inherit`

Reviewer 2: Please rename this to "be_inherited".

Author: Thanks, @clintoncwolfe. Did the required changes.

> `be_inherited` matcher returns the `Boolean`. It will return `true` if file or foler has inheritance enabled in Windows. This matcher only works for Windows OS.

Reviewer: ```suggestion
`be_inherited` is a boolean matcher which returns `true` if a file or folder has inheritance enabled, otherwise `false`. This matcher only works on Windows systems.
```
